### DISCUSSION TITLE
Suggested cycleway tag validation additions

### DISCUSSION THREAD

**Discussion by jonpsp** [openstreetmap/iD#9081] [Ideas]
Hello.

Can I suggest the following additional validation checks, please, all for highway=cycleway/path/footway?  These will help with clean-up of old data.

- Legacy tagging: if `cycleway=shared` is tagged, remove it and replace with `segregated=no`
- Legacy tagging: if `cycleway=segregated` is tagged, remove it and replace with `segregated=yes`
- Suspicious name: if `name` contains ` cycle path`, remove the text ` cycle path` (e.g. `name=North Street cycle path` would become `name=North Street`)

Jon
https://www.openstreetmap.org/user/jpennycook

**Comment by tordans** [comment]
Hi @jonpsp those sound like good changes to me but IMO likely better suited for other places than ID. AFAIK ID has little of those updates internally. However one of the QA tools integrated in iD should be a good fit for those changes. And also a MapRoulette Project - maybe even using the inline edit feature of MapRoulette (don't know what they call it from the top of my head). 

One thing that would make it easier for iD to add the changes is a documentation in the wiki that clearly tells users to upgrade those tags and marks them as deprecated.
